src/vmm/src/builder.rs

```diff
@@ -182,6 +189,21 @@ impl AsRawFd for SerialStdin {
     }
 }
 impl devices::legacy::ReadableFd for SerialStdin {}
+impl CustomData for SerialStdin {
```

Comment:
`CustomData` is not a very suggestive name, but I don't have a better idea for now. Maybe we can find something else by the time we merge this into master.

Reply:
+1

Reply:
Same here. I also don't have a better name in mind :(

Reply:
What about `VmmUserData` or `VmmRegistration`?

The name should hint that:
It is data that Vmm doesn't understand, but owns, and is therefore implicitly tied to the Vmm lifetime.

Reply:
In this case, `Opaque` could also work.

Reply:
Yes, `Opaque` is used in such cases. Should we go with simple `trait Opaque` or more descriptive `trait VmmOpaqueData`?

Reply:
I've renamed to `trait VmmEventsObserver` and updated comments and description accordingly.